src/XrdTpc/XrdTpcPMarkManager.cc

```diff
@@ -46,34 +46,38 @@ void PMarkManager::startTransfer(XrdHttpExtReq * req) {
 
 void PMarkManager::beginPMarks() {
   if(!mSocketInfos.empty() && mPmarkHandles.empty()) {
-    // Create the first pmark handle that will be used as a basis for the other handles
-    // if that handle cannot be created (mPmark->Begin() would return nullptr), then the packet marking will not work
-    // This base pmark handle will be placed at the beginning of the vector of pmark handles
+    // Create the first pmark handle
     std::stringstream ss;
     ss << "scitag.flow=" << mReq->mSciTag;
     SocketInfo & sockInfo = mSocketInfos.front();
-    mInitialFD = sockInfo.client.addrInfo->SockFD();
     std::unique_ptr<XrdNetPMark::Handle> initialPmark(mPmark->Begin(sockInfo.client, mReq->resource.c_str(), ss.str().c_str(), "http-tpc"));
     if(initialPmark) {
       // It may happen that the socket attached to the file descriptor is not connected yet. If this is the case the initial
```

Comment:
I don't see how this could happen as the socket is either connected in the call back or it's flagged as dead. When flagged as dead this code path shouldn't be activated should it? If it is, then the socket will never be connected and I suppose you can just simply ignore packet marking for it. In other words, this code seems superfluous with the change to connect the socket.